### ISSUE TITLE
[FEATURE] Ability to hide "Settings..."-Command in Submenu

### ISSUE THREAD

**Issue by 0nelight** [clechasseur/pathcopycopy#141]
I would like to be able to disable the "Settings..."-Command in the Submenu because I rarley edit the settings and therefore it would be nice to have an option to remove it from the Submenue to keep the UI clean. 
In case I disable it I can navigate to the Settings.exe and change my settings there.

**Comment by blackcrack** [comment]
this belongs to the extension and need to show ever, there is no possibility to unhide the settings agains.
Thing to the no-technical/non-extended Users, there want come massive troubles..
in direction "My settingsmenu is gone" "how can i recover my Settingsmenu" and so on..

bad idea... ! trust me.. i be more as 25 Years at some software .. 
(was also in Bulletin Board's over Terminal Emu. over Drt. with Cardware, Shareware and some Soft)
best
Blackcrack

Blackysgate .de / .net

**Comment by 0nelight** [comment]
@blackcrack We can offer the option in the registry so the possibility of accidentally disable this by a non-expert user can be avoided completely.

**Comment by n0f4c3** [comment]
I agree with blackcrack it's a very bad idea and even the way you suggest to set up the option is risky because an inexperienced user might decide to go and change the value in his registry and do the wrong thing

**Comment by 0nelight** [comment]
@n0f4c3 Your argument of "doing the wrong thing" is exactly what a user would do if he enables an option in the Windows-Registry he doesn't intend to enable - that's true for every bit in your computer... Only the users mind can protect him of doing that - and the barrier to do an unintended change of the windows registry is very high due all to the commands which have to be executed before we can change a registry-entry. Either you have to intentially play with settings or you have to be on some kind of drugs to do this unintentionally. I personally don't get your argument, but I am open to discuss.

**Comment by n0f4c3** [comment]
@ 0nelight me this is your argument several levels of security before you can access the regist I don't understand because there is only one and it is this window : ![20210503_083215](https://user-images.githubusercontent.com/14986315/116877381-ddb19880-abeb-11eb-80cf-a2c0e2c88aec.jpg)  (sorry for the French image I am Quebec Canadian)
so to come back to what I said he just has to click on yes and he can do whatever he wants even if it can be fatal for the computer so that's one of the reasons why your idea seems dangerous to me

**Comment by 0nelight** [comment]
What about messing with the million other registry-entries that can mess windows up? Would you touch them when you are not sure what you are doing? No you would not 😅 Additionally you need a lot of steps to change a specific registry entry when starting from the desktop. In case I don‘t answer anymore its because of the inaccuracies in your arguments - don‘t take it personal 😉

**Comment by n0f4c3** [comment]
@0nelight in the best case you are right but for a user who wants to experiment and learn he will do anything even if it can be dangerous because he does not know it

**Comment by n0f4c3** [comment]
@0nelight  me too I give up even if we were 2 trying to make you understand that your idea is not good because at least @blackcrack seems to have more experience than me in what he says so why don't you give up?

**Comment by clechasseur** [comment]
There's no need to argue all that much. We already have a setting to completely hide the submenu, which also prevents the user from accessing the settings, and it hasn't caused much trouble in all these years. I believe a registry-only entry would be fine.

**Comment by n0f4c3** [comment]
@clechasseur Thank you for your answer and sorry for all that

**Comment by clechasseur** [comment]
This has been implemented in `default` and will be part of the next release. To avoid displaying the Settings entry, add the following registry entry in `HKEY_CURRENT_USER\Software\clechasseur\PathCopyCopy`:

`AlwaysShowSettingsEntry` (type `DWORD`, value `0`)

**Comment by ggbbrr** [comment]
Wasn't this option already present before ?

Key = "KeyLock" ( 0 or 1 )

I know it still works.

**Comment by clechasseur** [comment]
> Wasn't this option already present before ?
> 
> Key = "KeyLock" ( 0 or 1 )
> 
> I know it still works.

`KeyLock` doesn't really do the same thing. While it doesn't display the Settings entry, true, it also disables editing settings altogether and prevents reading from settings in `HKEY_CURRENT_USER` if they exist.
